src/GovernorCountingFractional.sol

```diff
@@ -98,13 +98,17 @@ abstract contract GovernorCountingFractional is Governor {
         uint256 weight,
         bytes memory voteData
     ) internal virtual override {
-        require(!_proposalVotersHasVoted[proposalId][account], "GovernorCountingFractional: vote already cast");
-        _proposalVotersHasVoted[proposalId][account] = true;
+        require(weight > 0, "GovernorCountingFractional: no weight");
+        if (_proposalVotersWeightCast[proposalId][account] == weight) {
+          revert("GovernorCountingFractional: vote already cast");
+        }
```

Comment:
> Why not make this also a require for consistency?

It was a require and I removed it in 455cc1f4a2430f96decec6e4952faebbbc2f4185 because I found it confusing. I think this is more self-documenting.

> should we tweak the error string now that you can vote multiple times?

Sure, that's fair

Reply:
Error string tweaked in 575f596

Reply:
Fair enough, lgtm!